client/src/legacy/LeftAndMain.js

```diff
@@ -460,77 +460,93 @@ $.entwine('ss', function($) {
       // default to first button if none given - simulates browser behaviour
       if(!button) button = this.find('.btn-toolbar :submit:first');
 
-      form.trigger('beforesubmitform');
-      this.trigger('submitform', {form: form, button: button});
+      var beforeSubmitFormEventData = {
+        // array of promises that must resolve(true) before the form is submitted
+        promises: [],
+        // callbacks that are called on ajax success after submitted the form
+        onAjaxSuccessCallbacks: [],
+      };
+      form.trigger('beforesubmitform', beforeSubmitFormEventData);
+
+      Promise.all(beforeSubmitFormEventData.promises).then(function(results) {
+
+        for (const result of results) {
+          if (!result) {
```

Comment:
```suggestion
          if (result === false) {
```
I'd prefer this since that's [what you've said](https://github.com/silverstripe/silverstripe-admin/pull/1792#discussion_r1653845328) is the condition for this. Means we don't have to care about weird edge cases of other falsy values getting in here either with our own changes down the line or in peoples' customisations.